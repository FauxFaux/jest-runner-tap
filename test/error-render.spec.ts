import { testTranslation } from './sniff';

describe('error rendering', () => {
  it('handles exceptions', async () => {
    const result = await testTranslation('./fixtures/type-error.tap');
    expect(result.numFailingTests).toBe(1);
    expect(result.numPassingTests).toBe(1);
    expect(result.failureMessage).toMatchSnapshot();
  });
});
