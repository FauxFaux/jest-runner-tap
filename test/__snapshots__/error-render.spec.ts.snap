// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error rendering handles exceptions 1`] = `
"  âœ•  gonna throw
     {
       stack: [32m'buggy (type-error.js:14:3)\\\\nTest.<anonymous> (type-error.js:5:3)\\\\n'[39m,
       at: { line: [33m14[39m, column: [33m3[39m, file: [32m'type-error.js'[39m, function: [32m'buggy'[39m },
       type: [32m'TypeError'[39m,
       tapCaught: [32m'returnedPromiseRejection'[39m,
       test: [32m'gonna throw'[39m,
       source: [32m'function buggy(value) {\\\\n  value();\\\\n--^\\\\n}\\\\n'[39m
     }

"
`;
